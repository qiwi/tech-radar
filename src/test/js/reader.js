import {
  getReader,
  parse,
  parseCsvRadar,
  parseJsonRadar,
  parseYamlRadar
} from '../../main/js/parser/index.js'

describe('parser.js', () => {
  it('parse()', () => {
    expect(parse('src/test/stub/test.csv')).toMatchSnapshot()
  })
  it('csvReader', () => {
    expect(parseCsvRadar('src/test/stub/test.csv')).toMatchSnapshot()
  })
  it('jsonReader', () => {
    expect(parseJsonRadar('src/test/stub/test.json')).toMatchSnapshot()
  })
  it('yamlReader', () => {
    expect(parseYamlRadar('src/test/stub/test.yml')).toMatchSnapshot()
  })
  it('getReader ', () => {
    expect(getReader('.csv')).toBe(parseCsvRadar)
    expect(getReader('.json')).toBe(parseJsonRadar)
    expect(getReader('.yml')).toBe(parseYamlRadar)
  })
  it('invalid .csv', () => {
    expect(parse('src/test/stub/invalid.csv')).toStrictEqual({})
  })
})
