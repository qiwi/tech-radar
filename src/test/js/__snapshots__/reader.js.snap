// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser.js csvReader 1`] = `
Object {
  "data": Array [
    Object {
      "description": "Статически, типизированный ЖС",
      "moved": "1",
      "name": "TypeScript",
      "quadrant": "language",
      "ring": "Adopt",
    },
    Object {
      "description": "",
      "moved": "",
      "name": "Nodejs",
      "quadrant": "Platforms",
      "ring": "Adopt",
    },
    Object {
      "description": "Статический анализатор кода",
      "moved": "0",
      "name": "codeclimate",
      "quadrant": "tools",
      "ring": "Trial",
    },
    Object {
      "description": "Унификации контракта интерфейсов различных слоев приложений",
      "moved": "-1",
      "name": "Гексагональная архитектура",
      "quadrant": "Techniques",
      "ring": "Assess",
    },
  ],
  "meta": Object {
    "date": "2021-06-18",
    "legend": "bla bla bla",
    "title": "work it",
  },
  "quadrantAliases": Object {
    "language": "q1",
    "languages-and-frameworks": "q1",
    "platforms": "q2",
    "techniques": "q4",
    "tools": "q3",
  },
  "quadrantTitles": Object {
    "q1": "Languages and frameworks",
    "q2": "Platforms",
    "q3": "Tools",
    "q4": "Techniques",
  },
}
`;

exports[`parser.js jsonReader 1`] = `
Object {
  "data": Array [
    Object {
      "description": "Статически типизированный ЖС",
      "moved": "1",
      "name": "TypeScript",
      "quadrant": "languages-and-frameworks",
      "ring": "Adopt",
    },
    Object {
      "description": "",
      "moved": "",
      "name": "Nodejs",
      "quadrant": "Platforms",
      "ring": "Adopt",
    },
    Object {
      "description": "Статический анализатор кода",
      "moved": "0",
      "name": "codeclimate",
      "quadrant": "tools",
      "ring": "Trial",
    },
    Object {
      "description": "Унификации контракта интерфейсов различных слоев приложений",
      "moved": "-1",
      "name": "Гексагональная архитектура",
      "quadrant": "Techniques",
      "ring": "Assess",
    },
  ],
  "meta": Object {
    "date": "2021-06-12",
    "legend": "bla bla bla",
    "title": "tech radar js",
  },
  "quadrantAliases": Object {
    "languages-and-frameworks": "q1",
    "platforms": "q2",
    "techniques": "q4",
    "tools": "q3",
  },
  "quadrantTitles": Object {
    "q1": "Languages and frameworks",
    "q2": "Platforms",
    "q3": "Tools",
    "q4": "Techniques",
  },
}
`;

exports[`parser.js parse() 1`] = `
Object {
  "data": Array [
    Object {
      "description": "Статически, типизированный ЖС",
      "moved": 1,
      "name": "TypeScript",
      "quadrant": "q1",
      "ring": "adopt",
    },
    Object {
      "description": "",
      "moved": 0,
      "name": "Nodejs",
      "quadrant": "q2",
      "ring": "adopt",
    },
    Object {
      "description": "Статический анализатор кода",
      "moved": 0,
      "name": "codeclimate",
      "quadrant": "q3",
      "ring": "trial",
    },
    Object {
      "description": "Унификации контракта интерфейсов различных слоев приложений",
      "moved": -1,
      "name": "Гексагональная архитектура",
      "quadrant": "q4",
      "ring": "assess",
    },
  ],
  "meta": Object {
    "date": "2021-06-18",
    "legend": "bla bla bla",
    "title": "work it",
  },
  "quadrantAliases": Object {
    "language": "q1",
    "languages-and-frameworks": "q1",
    "platforms": "q2",
    "techniques": "q4",
    "tools": "q3",
  },
  "quadrantTitles": Object {
    "q1": "Languages and frameworks",
    "q2": "Platforms",
    "q3": "Tools",
    "q4": "Techniques",
  },
}
`;

exports[`parser.js yamlReader 1`] = `
Object {
  "data": Array [
    Object {
      "description": "Статически типизированный ЖС",
      "moved": 1,
      "name": "TypeScript",
      "quadrant": "languages-and-frameworks",
      "ring": "Adopt",
    },
    Object {
      "description": null,
      "moved": null,
      "name": "Nodejs",
      "quadrant": "Platforms",
      "ring": "Adopt",
    },
    Object {
      "description": "Статический анализатор кода",
      "moved": 0,
      "name": "codeclimate",
      "quadrant": "tools",
      "ring": "Trial",
    },
    Object {
      "description": "Унификации контракта интерфейсов различных слоев приложений",
      "moved": -1,
      "name": "Гексагональная архитектура",
      "quadrant": "Techniques",
      "ring": "Assess",
    },
  ],
  "meta": Object {
    "date": "2021-06-11",
    "legend": "bla bla bla",
    "title": "tech radar js",
  },
  "quadrantAliases": Object {
    "languages-and-frameworks": "q1",
    "platforms": "q2",
    "techniques": "q4",
    "tools": "q3",
  },
  "quadrantTitles": Object {
    "q1": "Languages and frameworks",
    "q2": "Platforms",
    "q3": "Tools",
    "q4": "Techniques",
  },
}
`;
